sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -90,6 +93,7 @@ def force_flush
               batch = snapshot.shift(@batch_size).map!(&:to_span_data)
               result_code = @exporter.export(batch)
               report_result(result_code, batch)
+              reset_on_fork(restart_thread: false) if @keep_running
```

Comment:
This should happen inside the `snapshot = lock { ... }` block, before `shift`ing the spans. We want to avoid accidentally sending buffered spans from the child.

Reply:
Thanks, have corrected it.